src/main/java/com/softdevsix/argos/service/ProjectValidator.java

```diff
@@ -0,0 +1,47 @@
+package com.softdevsix.argos.service;
+
+import com.softdevsix.argos.domain.Project;
+import com.softdevsix.argos.repository.ProjectRepository;
+import com.softdevsix.argos.repository.ProjectRulesRepository;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ProjectValidator {
+
+  private ProjectRepository projectRepository;
+  private ProjectRulesRepository projectRulesRepository;
+
+  @Autowired
+  public ProjectValidator(
+      ProjectRepository projectRepository, ProjectRulesRepository projectRulesRepository) {
+    this.projectRepository = projectRepository;
+    this.projectRulesRepository = projectRulesRepository;
+  }
+
+  public Project validate(Integer projectId) {
+    validateProjectAvailability(projectId);
+    return validateProjectExistence(projectId);
+  }
+
+  private void validateProjectAvailability(Integer projectId) {
+    projectRulesRepository
+        .findAll()
+        .forEach(
+            projectRules -> {
+              if (projectRules.getProject().getId().equals(projectId)) {
+                throw new IllegalArgumentException("Project already has rules");
+              }
+            });
+  }
+
+  private Project validateProjectExistence(Integer projectId) {
+    Optional<Project> optional = projectRepository.findById(projectId);
+    if (optional.isEmpty()) {
+      throw new IllegalArgumentException("Provided project ID is not registered");
```

Comment:
please create a Project creation exception, instead of using a generic exception, the new created exception should extend a RuntimeException